src/Entity/Customer/Identifier.php

```diff
@@ -22,31 +25,39 @@ class Identifier
      * Identifier constructor.
      *
      * @param string|int $id
+     * @throws InvalidEmail
      */
-    public function __construct($id, bool $isEmail = false)
+    public function __construct($id)
     {
-        $this->id      = $id;
-        $this->isEmail = $isEmail;
+        $this->isEmail = false;
+
+        if (!ctype_digit((string)$id)){
+            if ($this->validateEmail($id)) {
+                $this->isEmail = true;
+            } else {
+                throw new InvalidEmail();
+            }
+        }
+        $this->id = $id;
     }
 
-    /**
-     * @return int|string
-     */
-    public function getId()
+    public function getId(): string
     {
-        return $this->id;
+        return (string)$this->id;
```

Comment:
Se o ID não for um e-mail, a gente vai mandar ele como string lá para o customer.io? Não é interessante mandar ele como int quando for int?

Reply:
o id vai no query params, sempre vai ser uma string